### ISSUE TITLE
An option to "Show Notes instead of Activities" in current timeline

### ISSUE THREAD

**Issue by yvolk** [andstatus/andstatus#577]
As @vaschw suggested in https://github.com/andstatus/andstatus/issues/417#issuecomment-1444992367

> Is it possible to (optionally) change the sort order, so that the posts are sorted by post date rather than the date on which they got "liked" (or otherwise updated)?

Hi @vaschw! This thread ("issue") is many years old. Since that time AndStatus changed its internal data model largely. Mostly following ideas of Activity Streams and appeared later ActivityPub protocol.
Now "likes" or "retweets" DON'T UPDATE original message (called a "Note" in the modern terminology, see https://www.w3.org/TR/activitypub/ ).
Now most of timelines show Activities (actions), not "Notes". Thus sorting by a date of an Activity.

In this context your wish means an option to "Show Notes instead of Activities" in current timeline. With natural sorting by a date of a Note.

**Comment by yvolk** [comment]
@vaschw wrote:

> Yes, I think you have described the option I am asking for. In some cases it may be nice to view things in the order that the Activities happened (as the AndStatus Home Timeline currently does). In other cases, I may want the option to see the order that Notes were written in -- so that the date of a "like" does not change the ordering.

@vaschw You write about "cases". Do you mean that one time you will want to see a Timeline in the the Activities order, and another time you may decide to look **at the same Timeline** as a sequence of Notes (i.e. without "likes", "reposts", "follows"...) in the order of Notes.

**Comment by vaschw** [comment]
Yes, I might want to sort the same timeline differently at different times.  I realize now that "case" is a special programming term; my use of the word "case" was unintentional.  I just mean that I would like to have different options to choose from in the Timeline menu.

Personally, I would _usually_ like to view Notes in the order that they were posted, so that I can read them in order.  (It is desirable for there to still be an indication that a note has been "liked" or replied to, but I would prefer the _sort order_ to be done by Note post date.)  At other times I might like to sort the same timeline by the date of the Activity, in order to see which Notes have most recently been "liked" or replied to.

The interface could look like this: In the "Timeline" drop down menu, have e.g. an option for "Home - sorted by reaction date" and have a separate option for "Home - sorted by post date".  That language would make sense to me in the menu, but perhaps you have better idea.

Actually, I realize that if this is implemented as I just described then it will double the length of the "Timeline" dropdown menu.  So perhaps there should be a "sort by" toggle at the top of the Timeline dropdown menu (to choose between "sort by reaction date" and "sort by post date", so the option is easily accessible but will not double the number of items in the menu.

**Comment by yvolk** [comment]
Thank you for explanation. I know people may also like to see "Media timeline" as a scrolling list of images only. Maybe that could be the third timeline view option?!

Currently we have two independent timeline view options in the "Navigation drawer" on the left side of the timeline screen:
"Collapse duplicates" and "Show sensitive content" (see https://github.com/andstatus/andstatus/issues/547#issuecomment-951625975).
Maybe I need to add the new "View switch" there?!
